### ISSUE TITLE
Missing `token_type_ids` when different tokenizer and model

### ISSUE THREAD

**Issue by ZeusFSX** [huggingface/optimum#1758]
### System Info

```shell
optimum==1.17.1
```

### Who can help?

@philschmid @michaelbenayoun @JingyaHuang @echarlaix

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction (minimal, reproducible, runnable)

Hi I faced with a problem described [here](https://github.com/huggingface/optimum/issues/1500)  with missing `'token_type_ids'`. 

The root of problem is when we used tokenizer different from model, for `BERT` we don't need `attention_mask`, but we need `token_type_ids`. For example `intfloat/multilingual-e5-small` is BertModel architecture, but use XLMRobertaTokenizer, and we faced with problem missing 'token_type_ids' when we convert it onnx or openvino and try to run it with optimum.

For example: I finetuned `intfloat/multilingual-e5-small` for token-classification task and its works fine with transformer library. 

But when I try to use pipeline with ORTModelForTokenClassification, `token_type_ids` do not add in model input because tokenizer return `attention_mask` instead `token_type_ids`. 

And problem in code is [here](https://github.com/huggingface/optimum/blob/d87efb25c98741501fbf6da0d270fc181611b795/optimum/onnxruntime/modeling_ort.py#L1468) 

But when I added it manually, everything works: 
This fix will work:
```python
tokenizer = XLMRobertaTokenizerFast.from_pretrained('models/small')
model = ORTModelForTokenClassification.from_pretrained('models/small-onnx/')
inputs = tokenizer('some text', return_tensors='pt')
inputs['token_type_ids'] = inputs['attention_mask']
model(**inputs)
```

Maybe we should check the inputs of model and decide add `token_type_ids` or not, based on the input of model, not tokenizer output. Or we can do like in transformer lib. Here the [link](https://github.com/huggingface/transformers/blob/56b64bf1a51e29046bb3f8ca15839ff4d6a92c74/src/transformers/models/bert/modeling_bert.py#L973) 

They mask all `token_type_ids` with zeroes if they not preserve.

Also this problem is refer for openvino OVModelForTokenClassification.

### Expected behavior

Forward will be the same like in transformer lib
